recipe-server/client/actions/preference-experiment/index.js

```diff
@@ -31,38 +25,35 @@ export default class PreferenceExperimentAction extends Action {
     // Exit early if we're on an incompatible client.
     if (experiments === undefined) {
       this.normandy.log('Client does not support preference experiments, aborting.', 'info');
-      return Promise.resolve();
+      return;
     }
 
     seenExperimentNames.push(slug);
-    return experiments.has(slug).then(hasSlug => {
-      // If the experiment doesn't exist yet, enroll!
-      if (!hasSlug) {
-        return this.chooseBranch(branches).then(branch =>
-          experiments.start({
-            name: slug,
-            branch: branch.slug,
-            preferenceName,
-            preferenceValue: branch.value,
-            preferenceBranchType,
-            preferenceType,
-          })
-        );
-      }
-
-      // If the experiment exists, and isn't expired, bump the lastSeen date.
-      return experiments.get(slug).then(experiment => {
-        if (experiment.expired) {
-          this.normandy.log(`Experiment ${slug} has expired, aborting.`, 'debug');
-          return true;
-        }
 
-        return experiments.markLastSeen(slug);
+    // If the experiment doesn't exist yet, enroll!
+    const hasSlug = await experiments.has(slug);
+    if (!hasSlug) {
+      const branch = await this.chooseBranch(branches);
+      await experiments.start({
+        name: slug,
+        branch: branch.slug,
+        preferenceName,
+        preferenceValue: branch.value,
+        preferenceBranchType,
+        preferenceType,
       });
-    });
+    } else {
+      // If the experiment exists, and isn't expired, bump the lastSeen date.
+      const experiment = await experiments.get(slug);
+      if (experiment.expired) {
```

Comment:
I'm personally not a fan of `!(await whatever())`, if anything I'd break that into its own variable before the conditional statement, e.g.:

```
const ok = await whatever();
if (!ok) { ... }
```
In general, I agree with your thoughts on breaking actions into standalone functions. I think a lot of it can be determined case-by-case, and this is a small piece that it may not be necessary to refactor - what's there is fine for me. If you want to replace it with the alternative, that makes sense to me, too.